snuba/datasets/entities/discover.py

```diff
@@ -119,20 +119,16 @@ class DefaultIfNullFunctionMapper(FunctionCallMapper):
     called on NULL, change the entire function to be NULL.
     """
 
-    function_match = FunctionCallMatch(StringMatch("identity"), (LiteralMatch(),))
+    function_match = FunctionCallMatch(
+        StringMatch("identity"), (LiteralMatch(value=None, check_none_value=True),)
+    )
 
     def attempt_map(
         self,
         expression: FunctionCall,
         children_translator: SnubaClickhouseStrictTranslator,
     ) -> Optional[FunctionCall]:
 
-        # HACK: Quick fix to avoid this function dropping important conditions from the query
-        logical_functions = {"and", "or", "not", "xor"}
-
-        if expression.function_name in logical_functions:
-            return None
-
```

Comment:
Since this is a hack, I am trying to remove it from the code. The problem with https://github.com/getsentry/snuba/blob/master/snuba/datasets/entities/discover.py#L122 is that it matches any function like identity(*) where * could be anything. What we really want is to match identity(NULL). The LiteralMatch with the flag of check_none_value set to True would ensure that the parameter is of None type.

Reply:
I don't think that's the only thing this hack was doing.
The point was avoiding a function like 
`and(identity(....), equals(project_id, 1))` 
to be transformed into 
`identity(....)`
which would have lost the project_id condition which is a mandatory one.

Reply:
You could maybe remove the `not` but the other checks are here for an important reason. Otherwise this function would replace parts of the condition it shouldn't.